contracts/staged-contract-updates/StagedContractUpdates.cdc

```diff
@@ -354,19 +355,19 @@ access(all) contract StagedContractUpdates {
 
         /// Checks if the specified DelegatedUpdater Capability is contained and valid
         ///
-        access(all) fun check(id: UInt64): Bool? {
+        access(all) view fun check(id: UInt64): Bool? {
             return self.delegatedUpdaters[id]?.check() ?? nil
         }
 
         /// Returns the IDs of delegated Updaters
         ///
-        access(all) fun getUpdaterIDs(): [UInt64] {
+        access(all) view fun getUpdaterIDs(): [UInt64] {
             return self.delegatedUpdaters.keys
         }
 
         /// Allows for the delegation of contract updates as defined within the Updater resource
         ///
-        access(all) fun delegate(updaterCap: Capability<&Updater>) {
+        access(all) fun delegate(updaterCap: Capability<auth(UpdateContract) &Updater>) {
```

Comment:
I'm assuming that the functions in this resource don't need entitlements since they require the capability to call, but I just wanted to flag just in case

Reply:
That's right, only those that are currently entitled need to be privileged. `delegate` and `removeAsUpdater` can be public as they allow & remove delegation to this resource and those are necessarily called by the owner of the underyling `Updater` since the caps are entitled.